nservicebus/hosting/cloud-services-host/configuration.md

```diff
@@ -1,114 +1,78 @@
-      value="Endpoint=sb://{namespace here}.servicebus.windows.net/;
-      SharedAccessKeyName=RootManageSharedAccessKey;
-      SharedAccessKey={shared access key here}" />
-...
-    </ConfigurationSettings>
-  </Role>
-</ServiceConfiguration>
-```
-
-
-The "AzureServiceBusQueueConfig.ConnectionString" for Azure Service Bus namespace connection string can be retrieved from the Azure portal using an authorized account.
-
-In the NServiceBus solution, specify the endpoint configuration to use AzureServiceBus transport:
-
-```
-public class EndpointConfig : IConfigureThisEndpoint, AsA_Worker
-{
-    public void Customize(BusConfiguration builder)
-    {
-        builder.UseTransport<AzureServiceBusTransport>();
-    }
-}
-```
\ No newline at end of file
+## Configuration override convention
+
+Because Azure cloud services has it's own configuration model, but NServiceBus is typically used with it's configuration in app.config, a convention based configuration source is provided. Where most of the configuration is in app.config, but any setting can be added 'by convention' to the service configuration file to override the original value in app.config. This makes it easy to develop locally (without the service runtime), but still make use of this feature in production.
+
+NOTE: As most implementations for NServiceBus (like transports and persisters) are moving towards a code only configuration model, this feature will lose most of it's value over time.
+
+The configuration source can be turned on like this:
+
+snippet:AzureConfigurationSource
+
+This convention based override model works for all configuration sections used by NServicebus, but for the sake of example, let's use the `AzureServiceBusQueueConfig` section which is available in Azure Service Bus transport version 6 and below:
+
+Snippet:AzureServiceBusQueueConfigSection
+
+This setting can then be set in the app.config file, by specifying this config section:
+
+Snippet:AzureServiceBusQueueConfig
+
+When hosting in a Azure cloud service this setting can be overridden in the service configuration (`.cscfg`) file.
+
+First define the setting in the service definition file (`.csdef`).
+
+Snippet:AzureServiceBusQueueConfigCsDef
+
+Then specify the value for every cloud service deployment in the Cloud Services project.
+
+Snippet:AzureServiceBusQueueConfigCsCfg
+
+The name to be used for the property override is always structured like this: `TagName.PropertyName`. If tags were nested it's a repetition of the pattern `ParentTagName.ChildTagName.PropertyName`. It's currently impossible to override parent tags that would contain multiple of the same child tag, therefore it's impossible to override `MessageEndpointMappings` in this way.
+
+The override order used in this example applies, lowest priority is the default value set on the config section, then the app.config value is applied, and then the service configuration value is applied.
```

Comment:
> The override order used in this example applies, lowest priority is the default value set on the config section, then the app.config value is applied, and then the service configuration value is applied.

The default value set in the config section has the lowest priority. It can be overridden by the value specified in the `app.config` file. The value provided in the service configuration file takes precedence over the value specified in the `app.config` file.